### PR TITLE
Use native POSIX polling syscalls to read input

### PR DESCRIPTION
**Pull request by lvxnull**
This will decrease the amount of thread synchronization calls during input processing and memory usage by removing the dedicated input thread and moving input reading code into the main thread, in polling read mode(non-canon, VMIN and VTIME set to 0), as well as replace the manual, sleep based polling implementation with `pselect`. Because polling is now done by `pselect`, `SIGUSR1` will be used as the interrupt flag.
Additionally, input will be read in chunks instead of character by character.

### CODE REVIEW

**Comment by aristocratos** [comment]
@lvxnull 
This is great, have you tested it on all platforms?

But I do believe the interrupt flag will still be needed as a way to signal that the next call to `Input::poll` should immediately return false. Because the the new `Input::interrupt` function will only stop a currently ongoing call to `Input::poll` right?

**Comment by lvxnull** [comment]
@aristocratos 
I tested on linux only.
> the new Input::interrupt function will only stop a currently ongoing call to Input::poll right?

No, because `SIGUSR1` is only unblocked during `pselect`, and when a blocked signal is raised it stays in a pending state until it's unblocked and then executes immediately.

**Comment by aristocratos** [comment]
> No, because SIGUSR1 is only unblocked during pselect, and when a blocked signal is raised it stays in a pending state until it's unblocked and then executes immediately.

Neat 👍🏼 

Nice improvement over the previous mess!

Will do some stability testing on all platforms when I've got some time.

File: src/btop_input.cpp [hunk 1]
```diff
@@ -76,83 +73,46 @@ namespace Input {
...
-					current.clear();
-				}
-			}
-		}
-
-		size_t avail() {
-			std::lock_guard<std::mutex> g(lock);
-
-			return current.size();
+	bool poll(int timeout) {
```
Comment (nobounce) [inline src/btop_input.cpp:85]: Timeout can be `const`

**Comment by nobounce** [comment]
Also is there a reason to use `select()` over `poll()`?

File: src/btop_input.cpp [hunk 2]
```diff
@@ -76,83 +73,46 @@ namespace Input {
...
-			std::string res;
-
-			{
-				std::lock_guard<std::mutex> g(lock);
-
-				res.swap(current);
+		if(pselect(STDIN_FILENO + 1, &fds, nullptr, nullptr, waitptr, &signal_mask) > 0) {
+			input.clear();
+			char buf[1024];
+			int count = 0;
```
[the remark this answers is not in the document]
Comment (lvxnull) [reply src/btop_input.cpp:103]: I'll use `ssize_t` instead since it's the actual return type of `read()`

**Comment by lvxnull** [comment]
done

**Pull request merged**